### PR TITLE
Generated ACL in scope of ui component form generation

### PR DESCRIPTION
**Pull request by bohdan-harniuk**
**Description (*)**
This PR extends the new UI component form generation functionality with the generation of ACL resources XML file.

image
image
image
image

**Contribution checklist** (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with integration/functional tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: src/com/magento/idea/magento2plugin/util/magento/GetAclResourcesListUtil.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+package com.magento.idea.magento2plugin.util.magento;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.util.indexing.FileBasedIndex;
+import com.magento.idea.magento2plugin.stubs.indexes.xml.AclResourceIndex;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class GetAclResourcesListUtil {
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/util/magento/GetAclResourcesListUtil.java:15]: Util class should be final

File: src/com/magento/idea/magento2plugin/util/magento/GetAclResourcesListUtil.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    private static final String TEST_MODULE_ID_PART = "Magento_TestModule";
+    private static final GetAclResourcesListUtil INSTANCE = new GetAclResourcesListUtil();
+
+    /**
+     * Get GetAclResourcesListUtil instance.
+     *
+     * @return GetAclResourcesUtil
+     */
+    public static GetAclResourcesListUtil getInstance() {
+        return INSTANCE;
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/util/magento/GetAclResourcesListUtil.java:25]: Avoid using Singelton as util class

File: src/com/magento/idea/magento2plugin/util/magento/GetAclResourcesListUtil.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    /**
+     * Get acl resources list.
+     *
+     * @param project Project
+     *
+     * @return List
+     */
+    public List<String> execute(final Project project) {
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/util/magento/GetAclResourcesListUtil.java:35]: Util methods should be static

**Review by VitaliyBoyko — CHANGES_REQUESTED** [review]
Please add an integration test for the generator.

File: src/com/magento/idea/magento2plugin/util/magento/GetAclResourcesListUtil.java (hunk 1 above)
Comment (bohdan-harniuk) [reply src/com/magento/idea/magento2plugin/util/magento/GetAclResourcesListUtil.java:15]: Done

File: src/com/magento/idea/magento2plugin/util/magento/GetAclResourcesListUtil.java (hunk 3 above)
Comment (bohdan-harniuk) [reply src/com/magento/idea/magento2plugin/util/magento/GetAclResourcesListUtil.java:35]: Done

File: src/com/magento/idea/magento2plugin/util/magento/GetAclResourcesListUtil.java (hunk 2 above)
Comment (bohdan-harniuk) [reply src/com/magento/idea/magento2plugin/util/magento/GetAclResourcesListUtil.java:25]: Done

**Review by VitaliyBoyko — APPROVED** [review]

**Pull request merged**